### PR TITLE
Detaching ebs volume. Instance id from attachments array

### PR DESCRIPTION
**Pull request by nvivek91**
Instance id is present in attachments array of volume struct. Please refer at the documentation for the response format
http://docs.aws.amazon.com/sdkforruby/api/Aws/EC2/Client.html#describe_volumes-instance_method

### CODE REVIEW

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.chef.io/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## GitHub Users Who Are Not Authorized To Contribute

The following GitHub users do not appear to have signed a CLA:

* @nvivek91

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

File: providers/ebs_volume.rb [hunk 1]
```diff
@@ -240,12 +240,13 @@ def attach_volume(volume_id, instance_id, device, timeout)
 # Detaches the volume and blocks until done (or times out)
 def detach_volume(volume_id, timeout)
   vol = volume_by_id(volume_id)
-  if vol[:instance_id] != instance_id
-    Chef::Log.debug("EBS Volume #{volume_id} is not attached to this instance (attached to #{vol[:instance_id]}). Skipping...")
+  vol_instance_attachment = vol.attachments.detect{|attachment| attachment[:instance_id] == instance_id}
+  unless vol_instance_attachment
+    Chef::Log.debug("EBS Volume #{volume_id} is not attached to this instance (attached to #{instance_id}). Skipping...")
```
Comment (dhui) [inline providers/ebs_volume.rb]: log output is incorrect. The volume is not attached to `instance_id`

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.getchef.com/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## GitHub Users Who Are Not Authorized To Contribute

The following GitHub users do not appear to have signed a CLA:

* @nvivek91

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by tas50** [comment]
@nvivek91 I'm fairly sure we've fixed this issue on master now and it shipped in the 3.0 release.  I'm going to close this out now.  Let me know if you're still having issues.

**Pull request closed without merging**